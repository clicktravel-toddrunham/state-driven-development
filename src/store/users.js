const actionTypes = {
  FETCH_USERS: 'FETCH_USERS',
  SELECT_USER: 'SELECT_USER',
};

const stateTypes = {
  FETCHING_USERS: 'FETCHING_USERS',
  FETCHED_USERS: 'FETCHED_USERS',
  SELECTED_USER: 'SELECTED_USER',
};

const initialState = {
  stateType: stateTypes.FETCHING_USERS,
  users: [], // These could probably be in a 'data' object, may as well use this 
  user: [], // as an opportunity to give our stores more consistent structure
};

const fetchUsers = () => ({
  type: actionTypes.FETCH_USERS,
});

const selectUser = name => ({
  type: actionTypes.SELECT_USER,
  name,
});

const users = (state = initialState, action) => {
  switch (action.type) {
    case actionTypes.FETCH_USERS:
      return {
        ...state,
        stateType: stateTypes.FETCHED_USERS,
        users: [{ name: 'Tom' }, { name: 'Chris' }, { name: 'Sam' }],
      };
    case actionTypes.SELECT_USER:
      return {
        ...state,
        stateType: stateTypes.SELECTED_USER,
        user: state.users.filter(user => user.name === action.name),
      };
    default:
      return state;
  }
};

export { actionTypes, stateTypes, fetchUsers, selectUser, initialState };

export default users;
