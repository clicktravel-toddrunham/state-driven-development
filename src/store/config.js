import { createStore, combineReducers } from 'redux';

import users from './users';

const reducers = combineReducers({
  users,
});

const configureStore = () => createStore(reducers);

const configuredStore = configureStore();

export default configuredStore;
