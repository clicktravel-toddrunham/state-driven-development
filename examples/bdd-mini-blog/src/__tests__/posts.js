import React from 'react';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import Posts from '../containers/Posts';
import { initialState, stateTypes } from '../store/posts';

const mockStore = configureMockStore();

const posts = [
  {
    id: 1,
    title: 'Blog post one',
    content: 'Content here...',
    author: 'Chris',
  },
  {
    id: 2,
    title: 'Blog post two',
    content: 'Content here...',
    author: 'Sam',
  },
];
const post = [
  {
    id: 1,
    title: 'Blog post one',
    content: 'Content here...',
    author: 'Chris',
  },
];

describe('Posts container', () => {
  it('Scenario: The reader can view all posts', () => {
    const store = mockStore({
      posts: {
        ...initialState,
        _stateType: stateTypes.FETCHED_POSTS,
        data: { posts },
      },
    });
    const tree = renderer.create(<Posts store={store} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
