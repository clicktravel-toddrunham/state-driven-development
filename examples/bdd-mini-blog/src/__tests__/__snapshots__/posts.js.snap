// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts container Scenario: The reader can view all posts 1`] = `
<ul
  className="ui-grid list-group "
>
  <li
    className="pull-left"
  >
    <p
      className="pull-left"
      style={
        Object {
          "marginBottom": "15px",
          "width": "20%",
        }
      }
    >
      <span
        className="text-large link"
        onClick={[Function]}
      >
        1
      </span>
      
      
      <br />
      <span
        className="text-muted"
      >
        id
      </span>
    </p>
    <p
      className="pull-left"
      style={
        Object {
          "marginBottom": "15px",
          "width": "20%",
        }
      }
    >
      
      
      <span
        className="text-large"
      >
        Blog post one
      </span>
      <br />
      <span
        className="text-muted"
      >
        title
      </span>
    </p>
  </li>
  <li
    className="pull-left"
  >
    <p
      className="pull-left"
      style={
        Object {
          "marginBottom": "15px",
          "width": "20%",
        }
      }
    >
      <span
        className="text-large link"
        onClick={[Function]}
      >
        2
      </span>
      
      
      <br />
      <span
        className="text-muted"
      >
        id
      </span>
    </p>
    <p
      className="pull-left"
      style={
        Object {
          "marginBottom": "15px",
          "width": "20%",
        }
      }
    >
      
      
      <span
        className="text-large"
      >
        Blog post two
      </span>
      <br />
      <span
        className="text-muted"
      >
        title
      </span>
    </p>
  </li>
</ul>
`;
